pydantic/datetime_parse.py

```diff
@@ -81,6 +79,23 @@ def from_unix_seconds(seconds: Union[int, float]) -> datetime:
     return dt.replace(tzinfo=timezone.utc)
 
 
+def _parse_timezone(value: str, error: Callable[[], errors.PydanticValueError]) -> timezone:
```

Comment:
```suggestion
def _parse_timezone(value: str, error: Type[Exception]) -> Optional[timezone]:
```

Reply:
When the return annotation is 
-> Optional[timezone]:
The unittests all pass but calling mypy, I get this:
$ mypy pydantic/datetime_parse.py 
pydantic/datetime_parse.py:161: error: Incompatible types in assignment (expression has type "Optional[timezone]", target has type "Union[int, timezone]")  [assignment]
pydantic/datetime_parse.py:199: error: Incompatible types in assignment (expression has type "Optional[timezone]", target has type "Union[int, timezone]")  [assignment]
Found 2 errors in 1 file (checked 1 source file)